diff -u -r stats-1.0.3/config.w32 stats-1.0.3.patched/config.w32
--- stats-1.0.3/config.w32	2012-12-10 21:03:05.000000000 +0900
+++ stats-1.0.3.patched/config.w32	2014-05-14 12:57:35.585768504 +0900
@@ -4,6 +4,6 @@
 ARG_ENABLE("stats", "statistics support", "no");
 
 if (PHP_STATS != "no") {
-	EXTENSION('stats', 'php_stats.c com.c dcdflib.c ipmpar.c linpack.c randlib.c fd_e_lgamma_r.c fd_e_log.c fd_k_cos.c fd_k_sin.c fd_w_lgamma.c');
+	EXTENSION('stats', 'php_stats.c com.c dcdflib.c ipmpar.c linpack.c randlib.c e_lgamma_r.c e_log.c k_cos.c k_sin.c w_lgamma.c');
 }
 
diff -u -r stats-1.0.3/randlib.c stats-1.0.3.patched/randlib.c
--- stats-1.0.3/randlib.c	1997-08-21 04:02:32.000000000 +0900
+++ stats-1.0.3.patched/randlib.c	2014-05-14 12:57:09.698515052 +0900
@@ -5,7 +5,7 @@
 #define ABS(x) ((x) >= 0 ? (x) : -(x))
 #define min(a,b) ((a) <= (b) ? (a) : (b))
 #define max(a,b) ((a) >= (b) ? (a) : (b))
-void ftnstop(char*);
+extern void ftnstop(char*);
 float genbet(float aa,float bb)
 /*
 **********************************************************************
@@ -2093,13 +2093,3 @@
     return -num;
 else return num;
 }
-/************************************************************************
-FTNSTOP:
-Prints msg to standard error and then exits
-************************************************************************/
-void ftnstop(char* msg)
-/* msg - error message */
-{
-  if (msg != NULL) fprintf(stderr,"%s\n",msg);
-  exit(0);
-}
